sub/sd_lavc.c

```diff
@@ -472,7 +472,24 @@ static struct sub_bitmaps *get_bitmaps(struct sd *sd, struct mp_osd_res d,
             video_par = par;
     }
     if (priv->avctx->codec_id == AV_CODEC_ID_HDMV_PGS_SUBTITLE)
-        video_par = -1;
+    {
+        // For Blu-ray subs on SD video, try to match the video PAR.
```

Comment:
Do we have some reference that it is the correct thing to do? Or we are eyeballing it? I would prefer this comment to explain reasoning `why` and make it clear it is needed/correct. So we don't have to scratch our heads about it in few years in the future.

Reply:
I compared it to the output of an actual Blu-ray player in #15415. Would a comment about that issue number be enough?

Reply:
It's referenced in commit message, so that's enough. I'm more interested if we can produce some specification or reference to one about how this should be handled. I know blu-ray spec is not public, so even someone who knows this stuff to confirm.

I don't mind having this change, if it is limited to SD only. Just the guessing part if that's the correct and only resolution that we should do it for is iffy to me.

Reply:
There is the public white paper: https://digilabcfu.weebly.com/uploads/1/0/2/3/10237615/2b_bdrom_audiovisualapplication_0305-12955-15269.pdf

Under the "video streams" section all the HD resolutions have square pixels (except 1440x1080 but @cubicibo explained those in previous comments) and the two SD streams are the only ones with non-square pixels.

Reply:
Ok, so by restricting w/h we don't worry about cropped video and mismatch with subtitles, that would get stretched. I guess we can merge this, it has limited impact only to specific case.

Reply:
@kasper93 
> I'm more interested if we can produce some specification or reference to one about how this should be handled. I know blu-ray spec is not public, so even someone who knows this stuff to confirm.

The best answer I can give you is the following: the Blu-ray format always blend all planes in a standardized format (width, height, bit depth) (= space).
- HD content: PAR=1, mpv does not need to do any processing and just overlays as-is.
- SD content: the video and overlay are blended in the raw video format (SAR). mpv cannot overlay the Presentation Graphic at 1:1 due to PAR shenanigans.
Blu-ray only supports SD videos with storage sizes of 720x{480, 576}. Checking specifically for those is sufficient. Anything else is a mismatched video-subtitle pair, or HD assets, and can be safely assumed to require overlaying at a PAR=1.

If mpv ever supports Interactive Graphics Stream for menus, it will also need to perform this correction on SD footage.